csrc/quantization/fp8/amd/gemm_kernel.cu

```diff
@@ -0,0 +1,164 @@
+#include <cstdint>
+#include <cstdio>
+
+#include <torch/all.h>
+#include <c10/cuda/CUDAGuard.h>
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/cuda/CUDAContextLight.h>
+#include <hip/hip_runtime.h>
+#include <hipblaslt/hipblaslt.h>
+#include <hipblaslt/hipblaslt-ext.hpp>
+
+#define CHECK_CUDA(x) TORCH_CHECK(x.is_cuda(), #x " must be a CUDA tensor")
+#define CHECK_CONTIGUOUS(x) \
+  TORCH_CHECK(x.is_contiguous(), #x " must be contiguous")
+#define CHECK_INPUT(x) \
+  CHECK_CUDA(x);       \
+  CHECK_CONTIGUOUS(x)
+
+#ifndef CHECK_HIP_ERROR
+  #define CHECK_HIP_ERROR(error)                                    \
+    if (error != hipSuccess) {                                      \
+      fprintf(stderr, "Hip error: '%s'(%d) at %s:%d\n",             \
+              hipGetErrorString(error), error, __FILE__, __LINE__); \
+      exit(EXIT_FAILURE);                                           \
+    }
+#endif
+
+#ifndef CHECK_HIPBLASLT_ERROR
+  #define CHECK_HIPBLASLT_ERROR(error)                                  \
+    if (error != HIPBLAS_STATUS_SUCCESS) {                              \
+      fprintf(stderr, "hipBLASLt error: '%s'(%d) at %s:%d\n",           \
+              hipblasStatusToString(error), error, __FILE__, __LINE__); \
+      exit(EXIT_FAILURE);                                               \
+    }
+#endif
+
+/* TODO(HaiShaw): Have hipblasLt support mixed precision, s.t. input Tensors
+                  a and b can be float16 or bfloat16 type (performance A.I.)
+                  Extend interface to be more generic, to include bias, etc.
+ */
+void fp8_mm(torch::Tensor& a, torch::Tensor& b, torch::Tensor& result,
```

Comment:
please rename this `hipblas_scaled_mm` for consistency with our nvidia naming scheme:

`cutlass_scaled_mm`